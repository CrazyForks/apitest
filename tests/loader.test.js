const { spwanTest } = require("./utils");

describe("loader", () => {
  test("target file not found", async () => {
    const { stdout, code } = await spwanTest("loader/notfound.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("target main not found", async () => {
    const { stdout, code } = await spwanTest("loader", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid jsona", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-jsona.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid jsona value", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-jsona-value.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid client", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-client.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid mixin multiple", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-mixin-multiple.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid mixin jsona", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-mixin-jsona.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid mixin value", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-mixin-value.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid mixin", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-mixin.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid jslib js", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-jslib-js.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid jslib js2", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-jslib-js2.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid jslib", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-jslib.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid module jsona", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-module-jsona.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid module value", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-module-value.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid module", async () => {
    const { stdout, code } = await spwanTest("loader/invalid-module.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
});
