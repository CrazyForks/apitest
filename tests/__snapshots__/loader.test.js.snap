// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader invalid client 1`] = `
"main@client: should have object value at line 2 col 4
"
`;

exports[`loader invalid jslib 1`] = `
"[main@jslib] should have string value at line 2 col 4
"
`;

exports[`loader invalid jslib js 1`] = `
"main@jslib(invalid-jslib-404): load invalid-jslib-404.js failed
"
`;

exports[`loader invalid jslib js2 1`] = `
"main@jslib(invalid-jslib): load invalid-jslib.js throw Unexpected end of input at line 2 col 4
"
`;

exports[`loader invalid jsona 1`] = `
"main: load invalid-jsona.jsona throw unexpected token 'eof' at line 3 col 4
"
`;

exports[`loader invalid jsona value 1`] = `
"main: should have object value
"
`;

exports[`loader invalid mixin 1`] = `
"main@mixin: should have string value at line 2 col 4
"
`;

exports[`loader invalid mixin jsona 1`] = `
"main@mixin(invalid-jsona): load invalid-jsona.jsona throw unexpected token 'eof' at line 3 col 4
"
`;

exports[`loader invalid mixin multiple 1`] = `
"main@mixin: do not support multiple mixins at line 3 col 4
"
`;

exports[`loader invalid mixin value 1`] = `
"main@mixin(invalid-jsona-value): should have object value
"
`;

exports[`loader invalid module 1`] = `
"main@module: should have string value at line 2 col 4
"
`;

exports[`loader invalid module jsona 1`] = `
"main@module(invalid-jsona): load invalid-jsona.jsona throw unexpected token 'eof' at line 3 col 4
"
`;

exports[`loader invalid module value 1`] = `
"main@module(invalid-jsona-value): should have object value
"
`;

exports[`loader target file not found 1`] = `
"not found main jsona file
"
`;

exports[`loader target main not found 1`] = `
"not found main jsona file
"
`;
